packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -46,17 +53,67 @@ export class ResourceBtmService {
   constructor(
     private readonly gptService: GptService,
     private readonly prisma: PrismaService,
-    private readonly userActionService: UserActionService
+    private readonly userActionService: UserActionService,
+    private readonly billingService: BillingService,
+    private readonly analyticsService: SegmentAnalyticsService,
+    private readonly logger: AmplicationLogger
   ) {}
 
+  private async trackEvent(
+    user: User,
+    resource: Resource | ResourceDataForBtm,
+    eventName: EnumEventType,
+    customProperties: Record<string, unknown> = {}
+  ): Promise<void> {
+    try {
+      const subscription = await this.billingService.getSubscription(
+        user.workspace?.id
+      );
+
+      await this.analyticsService.track({
+        userId: user.id,
+        properties: {
+          ...customProperties,
+          workspaceId: user.workspace?.id,
+          projectId: resource.project?.id,
```

Comment:
please make sure these properties are sent from the calling functions

Reply:
The user and the resource are coming from the caller